pkg/apiserver/registry/stats/nodelatencystats/rest_test.go

```diff
@@ -13,3 +13,220 @@
+			summary: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+			nodeName: "node1",
+			expectedObj: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+			err: nil,
+		},
+		{
+			name: "delete summary not found",
+			summary: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+			nodeName:    "node2",
+			expectedObj: nil,
+			err:         errors.NewNotFound(statsv1alpha1.Resource("nodelatencystats"), "node2"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := NewREST()
+			ctx := context.Background()
+
+			_, err := r.Create(ctx, tt.summary, nil, nil)
+			require.NoError(t, err)
+			obj, deleted, err := r.Delete(ctx, tt.nodeName, nil, nil)
+			if tt.err != nil {
+				assert.EqualError(t, tt.err, err.Error())
+			} else {
+				require.NoError(t, err)
+				assert.True(t, deleted)
+				assert.Equal(t, tt.expectedObj, obj)
+			}
+		})
+	}
+}
+
+func TestRESTList(t *testing.T) {
+	summary := &statsv1alpha1.NodeLatencyStats{
+		ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+		PeerNodeLatencyStats: nil,
+	}
+	options := &internalversion.ListOptions{
+		Limit:    10,
+		Continue: "",
+	}
+	expectedObj := &statsv1alpha1.NodeLatencyStatsList{
+		Items: []statsv1alpha1.NodeLatencyStats{
+			{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+		},
+	}
+
+	r := NewREST()
+	ctx := context.Background()
+
+	_, err := r.Create(ctx, summary, nil, nil)
+	require.NoError(t, err)
+	objs, err := r.List(ctx, options)
+	require.NoError(t, err)
+	assert.Equal(t, expectedObj, objs)
+}
+
+func TestRESTConvertToTable(t *testing.T) {
+	mockTime := time.Date(2024, time.January, 1, 0, 0, 0, 0, time.UTC)
+	summary := &statsv1alpha1.NodeLatencyStats{
+		ObjectMeta: metav1.ObjectMeta{Name: "node1"},
+		PeerNodeLatencyStats: []statsv1alpha1.PeerNodeLatencyStats{
+			{
+				NodeName: "node2",
+				TargetIPLatencyStats: []statsv1alpha1.TargetIPLatencyStats{
+					{
+						TargetIP:                   "192.168.0.1",
+						LastSendTime:               metav1.Time{Time: mockTime},
+						LastRecvTime:               metav1.Time{Time: mockTime},
+						LastMeasuredRTTNanoseconds: 0,
+					},
+				},
+			},
+		},
+	}
+	expectedObj := &statsv1alpha1.NodeLatencyStats{
+		ObjectMeta: metav1.ObjectMeta{Name: "node1"},
+		PeerNodeLatencyStats: []statsv1alpha1.PeerNodeLatencyStats{
+			{
+				NodeName: "node2",
+				TargetIPLatencyStats: []statsv1alpha1.TargetIPLatencyStats{
+					{
+						TargetIP:                   "192.168.0.1",
+						LastSendTime:               metav1.Time{Time: mockTime},
+						LastRecvTime:               metav1.Time{Time: mockTime},
+						LastMeasuredRTTNanoseconds: 0,
+					},
+				},
+			},
+		},
+	}
```

Comment:
I don't think defining this object is useful, and I don't think the `assert.Equal(t, expectedObj, obj.Rows[0].Object.Object)` is very useful either.

Reply:
Okay, the two parts are really the same and can be left out of the comparison.